File: Creational - Abstract Factory/Typescript/concrete-factory/concrete-factory-1.ts
import { AbstractFactory } from '../abstract-factory/abstract-factory.ts'
import { AbstractProductA } from '../abstract-product/abstract-product-a.ts';
import { AbstractProductB } from '../abstract-product/abstract-product-b.ts';
import { ConcreteProductA1 } from '../concrete-product/concrete-product-a1.ts';
import { ConcreteProductB1 } from '../concrete-product/concrete-product-b1.ts';

export class ConcreteFactory1 implements AbstractFactory {
  public createProductA(): AbstractProductA {
      return new ConcreteProductA1();
  }

  public createProductB(): AbstractProductB {
      return new ConcreteProductB1();
  }
}

File: Creational - Abstract Factory/Python/concrete_factory/concrete_factory_1.py
from abstract_factory.abstract_factory import AbstractFactory
from abstract_product.abstract_product_a import AbstractProductA
from abstract_product.abstract_product_b import AbstractProductB
from concrete_product.concrete_product_a1 import ConcreteProductA1
from concrete_product.concrete_product_b1 import ConcreteProductB1


class ConcreteFactory1(AbstractFactory):
    def create_product_a(self) -> AbstractProductA:
        return ConcreteProductA1()

    def create_product_b(self) -> AbstractProductB:
        return ConcreteProductB1()